### PR TITLE
ADBDEV-4907-49: Add check for GetFileSegInfo result

### PR DESCRIPTION
**Pull request by RekGRpth**
Add check for GetFileSegInfo result

The AppendOnlyCollectDeadSegments function does not check what the GetFileSegInfo
function returns. Because of this, a null pointer dereference was possible. This
patch adds such a check.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/63343

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1035063

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1035064

**Comment by BenderArenadata** [comment]
Failed job Regression tests with Postgres on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1035056

File: src/backend/access/appendonly/appendonly_compaction.c [hunk 1]
```diff
@@ -519,8 +520,10 @@ AppendOnlyCollectDeadSegments(Relation aorel, List *compaction_segno)
 	Assert(Gp_role == GP_ROLE_EXECUTE || Gp_role == GP_ROLE_UTILITY);
 	Assert(RelationIsAoRows(aorel));
 
+	relname = RelationGetRelationName(aorel);
```
Comment (bandetto) [inline src/backend/access/appendonly/appendonly_compaction.c:523]: Does this produce a compilation warning without explicitly casting `char *` to `const char *`?

Comment (RekGRpth) [reply src/backend/access/appendonly/appendonly_compaction.c:523]: I did as [in](https://github.com/arenadata/gpdb/pull/695/files#diff-4d241dcb0deba3493d7dfa8c8488e87e032194e003373bf9b48bfe8685c6f4faR615-R633)

**Review by bandetto — APPROVED** [review]

**Review by bandetto — APPROVED** [review]

**Pull request merged**